### DISCUSSION TITLE
Check if user has a reaction in a foreach loop

### DISCUSSION THREAD

**Discussion by Xoshbin** [maize-tech/laravel-markable#16] [Q&A] [answered]
Hi, Thank you for the great package first.
I was able to check if the current user liked or had a reaction to an article (Article model), with this code in the documentation:
`Reaction::has($post, $user, 'heart'); // returns whether the user has reacted with the 'heart' reaction to the given post or not
`
That's for one model.
But what if I want to check it in a for each loop.

I have comments in the Article mode, Added Reactions to the Comment model. I want to be able to check if current user liked any comment or no. I tried in some ways but so far got no luck.

What I have tried so far:
```
@foreach($article->comments as $comment)
{{ $comment->reactions()->has('like') }}
@endforeach
```

and 

```
@foreach($article->comments as $comment)
{{ $comment->reactions()->where('user_id', Auth::id()) == true ) }}
@endforeach
```

**Comment by Xoshbin** [comment, marked as answer]
After asking the question I saw this https://github.com/maize-tech/laravel-markable/discussions/10
Sorry 😢

**Reply by riccardodallavia** [reply]
Hi @Xoshbin !

Yes, that's exactly how you can achieve it! 😛

Hope you like the solution, please let me know if you see any issues on this 😄

**Reply by Xoshbin** [reply]
Hi @riccardodallavia 
Thank you the problem is solved thanks.
But I was wondering if I may be able to access it somehow like this
```
@foreach($article->comments as $comment)
{{ $comment->reactions()->has('like') }}
@endforeach
```

**Comment by Xoshbin** [comment]
Hi @riccardodallavia I tried that way it's very heavy, for 15 records of articels it makes around 30 queries.
I tried to decrease the N+1 queries.
Is there any better way, please
Thank you

**Reply by riccardodallavia** [reply]
Hi @Xoshbin !

One way might be by using the `withExists` method in your query:

```
$comments = Comment::query()
  ->withExists([
    'reactions as has_reaction' => fn($query) => $query->where('user_id', auth()->id())
  ])
  ->get();
```

this way you could cycle through the comments and access the has_reaction attribute:

```
@foreach($comments as $comment)
{{ $comment->has_reaction }}
@endforeach
```
